merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/postgres/PostgresProfileQueryHandler.java

```diff
@@ -0,0 +1,190 @@
+package gov.nasa.jpl.aerie.merlin.worker.postgres;
+
+import gov.nasa.jpl.aerie.json.JsonParser;
+import gov.nasa.jpl.aerie.merlin.driver.resources.ResourceProfile;
+import gov.nasa.jpl.aerie.merlin.driver.resources.ResourceProfiles;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.RealDynamics;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.DatabaseException;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.FailedInsertException;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.FailedUpdateException;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PreparedStatements;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+
+import static gov.nasa.jpl.aerie.merlin.driver.json.SerializedValueJsonParser.serializedValueP;
+import static gov.nasa.jpl.aerie.merlin.server.http.ProfileParsers.realDynamicsP;
+import static gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresParsers.discreteProfileTypeP;
+import static gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresParsers.realProfileTypeP;
+
+/**
+ * Utility class to handle upload of resource profiles to the database. This class allows for each upload to have its
+ * own Connection object.
```

Comment:
Nonblocking Suggestion: remove the second part of this class comment, since "each upload having its own connection object" isn't an inherit property of the class (since the same instance can work for multiple uploads):
```suggestion
 * Utility class to handle upload of resource profiles to the database.
```

Reply:
Good idea. Implemented in 24221cb51036a46a1dc8b4a7774a9e033073279e.